ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -614,8 +619,10 @@ func (r *RayClusterReconciler) reconcileHeadlessService(ctx context.Context, ins
 func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv1.RayCluster) error {
 	logger := ctrl.LoggerFrom(ctx)
 
-	// if RayCluster is suspended, delete all pods and skip reconcile
-	if instance.Spec.Suspend != nil && *instance.Spec.Suspend {
+	// if RayCluster is suspending, delete all pods and skip reconcile
+	suspendStatus := utils.FindRayClusterSuspendStatus(instance)
+	if suspendStatus == rayv1.RayClusterSuspending ||
+		(!features.Enabled(features.RayClusterStatusConditions) && instance.Spec.Suspend != nil && *instance.Spec.Suspend) {
 		if _, err := r.deleteAllPods(ctx, common.RayClusterAllPodsAssociationOptions(instance)); err != nil {
```

Comment:
Do we need to send a `deleteAllPods` request to the K8s API server for each reconciliation during `Suspending`?

Reply:
Yes, we do that until all pods are deleted and then we switch to `suspended`.

Reply:
After revising the source, I found that `deleteAllPods` already did the improvement: only sends API requests if there are Pods still running.
https://github.com/ray-project/kuberay/blob/35e913a9b96e84b8e4a41ebe1efc6e656420afbd/ray-operator/controllers/ray/raycluster_controller.go#L195-L210

I guess there is no need for an issue to track the progress?